src/components/textToSpeech/js/TextToSpeech.js

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2015 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+// Declare dependencies
+/* global speechSynthesis, SpeechSynthesisUtterance*/
+
+var fluid_2_0 = fluid_2_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    fluid.registerNamespace("fluid.textToSpeech");
+
+    fluid.textToSpeech.isSupported = function () {
+        return !!(window && window.speechSynthesis);
+    };
+
+    fluid.defaults("fluid.textToSpeech", {
+        gradeNames: ["fluid.eventedComponent", "autoInit"],
+        events: {
+            start: null,
+            end: null,
+            error: null,
+            pause: null,
+            resume: null,
+            mark: null,
+            boundary: null
+        },
+        utteranceOpts: {
+            // text: "", // text to synthesize. avoid as it will override any other text passed in
```

Comment:
yes, these are all optional. And depending on the synthesizer used could have different results.. For example, volume and rate may or may not be respected by the synthesizer. Specifying the text would override any text actually passed in. so this one is a bit dangerous to use. Unless you really have a specific use case for it i guess.